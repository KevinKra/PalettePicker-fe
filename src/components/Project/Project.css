.Project {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  border-top: 1px solid coral;
  border-bottom: 5px solid coral;
  margin-bottom: 0.2rem;
  background-color: white;
  z-index: 1;
  min-height: 25vh;
}

.project-title {
  position: absolute;
  z-index: 2;
  top: -9%;
  left: 2%;
  padding: 0.5rem 2rem;
  background-color: white;
  transform: skew(35deg) rotate(1.5deg);
  background-color: coral;
}

.dotted-line {
  position: absolute;
  z-index: 1;
  top: -45%;
  width: 92%;
  height: 100px;
  border-bottom: 1px dashed coral;
}

.delete-wrapper {
  position: absolute;
  z-index: 2;
  top: -9%;
  right: 2%;
  border: 2px solid coral;
  background-color: coral;
  padding: 1rem 0.3rem 0.3rem 0.3rem;
  transform: skew(-15deg);
}

.project-delete {
  cursor: pointer;
  border: 2px solid rgb(40, 48, 56);
  background-color: rgb(40, 48, 56);
  padding: 0.5rem 0.7rem;
  transform: skew(-5deg);
}

.project-delete > h2 {
  pointer-events: none;
  margin: 0;
  color: white;
  font-size: 16px;
  transform: skew(2deg);
}

.project-title > h2 {
  margin: 0;
  font-family: "Carter One", cursive;
  font-weight: 100;
  transform: skew(-35deg);
  color: white;
}
