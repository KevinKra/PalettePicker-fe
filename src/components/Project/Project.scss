.Project {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
  border-top: 1px solid coral;
  border-bottom: 5px solid coral;
  margin-bottom: 0.2rem;
  background-color: white;
  z-index: 1;
  min-height: 25vh;
}

.project-title {
  position: absolute;
  z-index: 2;
  top: -1.1rem;
  left: 2.4rem;
  padding: 0.5rem 2rem;
  background-color: white;
  transform: skew(35deg) rotate(1.5deg);
  background-color: coral;
}

.dotted-line {
  position: absolute;
  z-index: 0;
  top: 0.5rem;
  width: 100%;
  height: 100px;
  border-top: 1px dashed coral;
}

.delete-wrapper {
  position: absolute;
  z-index: 2;
  top: -1.3rem;
  right: 2rem;
  border: 2px solid coral;
  background-color: coral;
  padding: 1rem 0.3rem 0.3rem 0.3rem;
  transform: skew(-15deg);
}

.project-delete {
  cursor: pointer;
  border: 2px solid rgb(40, 48, 56);
  background-color: rgb(40, 48, 56);
  padding: 0.5rem 0.7rem;
  transform: skew(-5deg);
}

.project-delete > h2 {
  pointer-events: none;
  margin: 0;
  color: white;
  font-size: 16px;
  transform: skew(2deg);
}

.project-title > h2 {
  margin: 0;
  font-family: "Carter One", cursive;
  font-weight: 100;
  transform: skew(-35deg);
  color: white;
}
