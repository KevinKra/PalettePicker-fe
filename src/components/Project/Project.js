import React from "react";
import Palette from "../Palette/Palette";
import { TweenMax } from "gsap/all";
import { Transition } from "react-transition-group";
import "./Project.scss";

const Project = props => {
  const palettes = props.palettes.map(palette => {
    return (
      <Palette
        title={palette.paletteTitle}
        colors={palette.paletteColors}
        key={palette.paletteId}
      />
    );
  });

  const { in: show, removeProject, project } = props;
  return (
    <Transition
      timeout={1000}
      mountOnEnter
      unmountOnExit
      appear
      in={show}
      addEndListener={(node, done) => {
        TweenMax.to(node, 0.55, {
          y: 0,
          autoAlpha: show ? 1 : 0,
          onComplete: done,
          delay: !show ? 0 : project.init ? props.index * 0.15 : 0
        });
      }}
    >
      <section className="Project">
        <div className="project-title">
          <h2>{props.title}</h2>
        </div>
        <div className="dotted-line" />
        <div className="delete-wrapper">
          <button
            onClick={() => removeProject(props.id)}
            className="project-delete"
          >
            <h2>X</h2>
          </button>
        </div>
        {palettes}
      </section>
    </Transition>
  );
};

export default Project;
