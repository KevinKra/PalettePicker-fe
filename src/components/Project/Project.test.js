import React from "react";
import Project from "./Project";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("Project", () => {
  let wrapper;
  const mockPalette = [{ paletteTitle: "ItsMe!", paletteColors: ["#111"] }];
  const removeProjectMock = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <Project palettes={mockPalette} removeProject={removeProjectMock} />
    );
  });
  it("should match the snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should invoke the 'removeProject' function when '.project-delete' is called", () => {
    const button = wrapper.find(".project-delete");
    button.simulate("click");
    expect(removeProjectMock).toHaveBeenCalled();
  });
});
