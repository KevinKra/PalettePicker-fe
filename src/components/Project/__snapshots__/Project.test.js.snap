// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match the snapshot 1`] = `
<Transition
  addEndListener={[Function]}
  appear={true}
  enter={true}
  exit={true}
  in={false}
  mountOnEnter={true}
  onEnter={[Function]}
  onEntered={[Function]}
  onEntering={[Function]}
  onExit={[Function]}
  onExited={[Function]}
  onExiting={[Function]}
  timeout={1000}
  unmountOnExit={true}
>
  <section
    className="Project"
  >
    <div
      className="project-title"
    >
      <h2 />
    </div>
    <div
      className="dotted-line"
    />
    <div
      className="delete-wrapper"
    >
      <button
        className="project-delete"
        onClick={[Function]}
      >
        <h2>
          X
        </h2>
      </button>
    </div>
    <withRouter(Palette)
      colors={
        Array [
          "#111",
        ]
      }
      key="NaN"
      title="ItsMe!"
    />
  </section>
</Transition>
`;
